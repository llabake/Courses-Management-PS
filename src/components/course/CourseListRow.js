/* eslint-disable jsx-a11y/no-noninteractive-element-interactions */
/* eslint-disable jsx-a11y/click-events-have-key-events */
import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

const CourseListRow = ({ course, onDelete }) => {
  const deleteCourse = () => (
    onDelete(course.id)
  );
  return (
    <tr>
      <td><a href={`/course/${course.watchHref}`} target="_blank" rel="noopener noreferrer">Watch</a></td>
      <td><Link to={`/course/${course.id}`}>{course.title}</Link></td>
      <td>{course.authorId}</td>
      <td>{course.category}</td>
      <td>{course.length}</td>
      <td
        className="btn btn-xs btn-danger"
        onClick={deleteCourse}
      >
Delete
      </td>
    </tr>
  );
};

CourseListRow.propTypes = {
  course: PropTypes.object.isRequired,
  onDelete: PropTypes.func.isRequired,
};

export default CourseListRow;
