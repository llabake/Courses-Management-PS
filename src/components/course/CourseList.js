import React from 'react';
import PropTypes from 'prop-types';
import CourseListRow from "./CourseListRow";

const CourseList = ({ courses, onDelete }) => (
  <table className="table">
    <thead>
      <tr>
        <th>&nbsp;</th>
        <th>Title</th>
        <th>Author</th>
        <th>Category</th>
        <th>Length</th>
        <th>Delete</th>
      </tr>
    </thead>
    <tbody>
      {courses.map(course => (
        <CourseListRow
        key={course.id}
        course={course}
        onDelete={onDelete} />
      ))}
    </tbody>
  </table>
);

CourseList.propTypes = {
  courses: PropTypes.array.isRequired,
};

export default CourseList;
