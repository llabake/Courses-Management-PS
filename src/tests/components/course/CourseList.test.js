import React from 'react';
import { shallow, mount } from 'enzyme';
import CourseList from "../../../components/course/CourseList";
import courseMock from "../../mocks/courseMock";
import CourseListRow from "../../../components/course/CourseListRow";

describe('<CourseList />', () => {
  it('renders correctly', () => {
    const props = {
      onDelete: jest.fn(),
      courses: courseMock.courses,
    };
    const wrapper = shallow(<CourseList {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
  it('renders courseList Row correctly ', () => {
    const props = {
      onDelete: jest.fn(),
      courses: courseMock.courses,
    };
    const wrapper = shallow(<CourseList {...props} />);
    expect(wrapper.find(CourseListRow).length).toBe(5);
  });
});
