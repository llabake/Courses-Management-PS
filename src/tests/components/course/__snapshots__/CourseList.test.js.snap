// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CourseList /> renders correctly 1`] = `
<table
  className="table"
>
  <thead>
    <tr>
      <th>
        Â 
      </th>
      <th>
        Title
      </th>
      <th>
        Author
      </th>
      <th>
        Category
      </th>
      <th>
        Length
      </th>
      <th>
        Delete
      </th>
    </tr>
  </thead>
  <tbody>
    <CourseListRow
      course={
        Object {
          "authorId": "cory-house",
          "category": "JavaScript",
          "id": "react-flux-building-applications",
          "length": "5:08",
          "title": "Building Applications in React and Flux",
          "watchHref": "http://www.pluralsight.com/courses/react-flux-building-applications",
        }
      }
      key="react-flux-building-applications"
      onDelete={[MockFunction]}
    />
    <CourseListRow
      course={
        Object {
          "authorId": "cory-house",
          "category": "Software Practices",
          "id": "clean-code",
          "length": "3:10",
          "title": "Clean Code: Writing Code for Humans",
          "watchHref": "http://www.pluralsight.com/courses/writing-clean-code-humans",
        }
      }
      key="clean-code"
      onDelete={[MockFunction]}
    />
    <CourseListRow
      course={
        Object {
          "authorId": "cory-house",
          "category": "Software Architecture",
          "id": "architecture",
          "length": "2:52",
          "title": "Architecting Applications for the Real World",
          "watchHref": "http://www.pluralsight.com/courses/architecting-applications-dotnet",
        }
      }
      key="architecture"
      onDelete={[MockFunction]}
    />
    <CourseListRow
      course={
        Object {
          "authorId": "cory-house",
          "category": "Career",
          "id": "career-reboot-for-developer-mind",
          "length": "2:30",
          "title": "Becoming an Outlier: Reprogramming the Developer Mind",
          "watchHref": "http://www.pluralsight.com/courses/career-reboot-for-developer-mind",
        }
      }
      key="career-reboot-for-developer-mind"
      onDelete={[MockFunction]}
    />
    <CourseListRow
      course={
        Object {
          "authorId": "daniel-stern",
          "category": "React",
          "id": "testing-react-applications-with-jest",
          "length": "5:10",
          "title": "Testing React Application With Jest",
          "watchHref": "http://www.pluralsight.com/courses/testing-react-applications-with-jest",
        }
      }
      key="testing-react-applications-with-jest"
      onDelete={[MockFunction]}
    />
  </tbody>
</table>
`;
